consul/util.go

```diff
@@ -153,10 +153,37 @@ func isPrivateIP(ip_str string) bool {
 	return false
 }
 
+// Returns addresses from interfaces that is up and supports
+// both broadcast and multicast
+func activeInterfaceAddresses() ([]net.Addr, error) {
+	var out []net.Addr
+
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get interfaces: %v", err)
+	}
+
+	reqFlags := net.FlagUp | net.FlagBroadcast | net.FlagMulticast
```

Comment:
We don't actually require Broadcast or Multicast support. Any reason to check for it? Also how does this method affect Windows?